mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/TokenDetailScreen.tsx

```diff
@@ -82,12 +95,23 @@ export function TokenDetailScreen ({
   route,
   navigation
 }: Props): JSX.Element {
+  const { denominationCurrency } = useDenominationCurrency()
```

Comment:
Do we need all of these to be here? Maybe instead of fetching it here, we can get them from the component itself (if it's only used there). To prevent passing lots of props

Reply:
We need it as it's not passed from any parent component

Reply:
I mean, instead of passing it from parent component. Maybe we can just call these data inside the component. If it's only used there

Reply:
Will call hook in children instead of passing props